src/nvim/buffer.c

```diff
@@ -3407,6 +3407,13 @@ typedef enum {
 } NumberBase;
 
 
+static size_t stl_items_len = 20;  // Initial value, grows as needed.
+static int *stl_groupitems = NULL;
+static stl_item_t *stl_items = NULL;
+static stl_hlrec_t *stl_hltab = NULL;
+static StlClickRecord *stl_tabtab = NULL;
+static int *stl_separator_locations = NULL;
```

Comment:
Your Vim patch puts these static variables here instead of within `build_stl_str_hl`. Do you intend to use these in other functions?

Reply:
Does it? As far as I can see in both my vim PR and the corresponding vim patch they are outside. They were there because they were needed there but they could come in in this context. Should I take them in?

Reply:
I went ahead and took them in.